src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexRunner.cs

```diff
@@ -198,6 +204,8 @@ public abstract class RegexRunner
                 // We failed to match at this position.  If we're at the stopping point, we're done.
                 if (runtextpos == stoppos)
                 {
+                    runtext = null; // drop reference to text to avoid keeping it alive in a cache
+                    if (runmatch != null) runmatch.Text = null!;
```

Comment:
`null!` means "ignore non nullability of `Text`" ? That makes it hard to reason about within that type?

Reply:
The alternative is to mark Text as nullable and then use ! everywhere we use it.  Would you prefer that?

Reply:
That might be preferable since it at least means that within the type, in places that you don't use a `!` there is a possibility the compiler might warn you of a genuine issue. Whereas if we set null externally, it might almost be better to just disable nullable on the file as we probably have `!` on almost everything.

Reply:
This is what we're talking about:
https://github.com/stephentoub/runtime/commit/7695854490fd4122890cbf7d65f6d53a03b9aca2
You'd prefer that?  I see the current nulling out as being equivalent to when we null something out in a Dispose.